x/evm/handler.go

```diff
@@ -73,22 +73,27 @@ func handleETHTxMsg(ctx sdk.Context, k Keeper, msg types.EthereumTxMsg) sdk.Resu
 	k.CommitStateDB.Prepare(ethHash, common.Hash{}, k.TxCount.Get())
 	k.TxCount.Increment()
 
-	bloom, res := st.TransitionCSDB(ctx)
-	if res.IsOK() {
-		k.Bloom.Or(k.Bloom, bloom)
+	// TODO: move to keeper
+	bloom, res, err := st.TransitionCSDB(ctx)
+	if err != nil {
+		return nil, err
 	}
-	return res
+
+	// TODO: why is this necessary ?
```

Comment:
just going to copy what I sent you for reference:
"It’s needed to update the block bloom filter. There is a bloom filter for each transaction which shows up in the transaction receipt, and also one for the block (which is just a bit or of all the transactions bloom filters in that block)"